options.go

```diff
@@ -50,8 +53,12 @@ func RegisterCommandLineOptions(flags *flag.FlagSet) (opts *HmacProxyOpts) {
 	return
 }
 
+// Validate ensures that the HmacProxyOpts configuration is correct and parses
+// some of the values into a useable format. It also sets the Mode member that
+// determines which proxy handler to launch. Collects as many error messages
+// as possible and returns them as a single string via the err return value.
 func (opts *HmacProxyOpts) Validate() (err error) {
-	msgs := make([]string, 0)
+	var msgs []string
```

Comment:
Hmm, the link actually seems to argue in favor of the `var msgs []string` format... And I switched to this format based on the output of [`golint`](https://github.com/golang/lint):
```
options.go:61:2: can probably use "var msgs []string" instead
```

At any rate, the `var msgs []string` version passes all the tests. :-)

Also, it seems the slice structure is cheap enough to pass by value in this case, and passing by value appears to be the idiomatic use throughout the standard library. The `append()` built-in also takes the original slice as an argument, then returns a new slice, as is happening with the `validate*()` functions here. Some more background:

https://blog.golang.org/go-slices-usage-and-internals
https://golang.org/doc/effective_go.html#slices

So, great discussion, but I'm inclined to stick with the status quo.